classes/class-enqueues.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Enqueue assets.
+ *
+ * @package Advanced_Multi_Block
+ */
+
+namespace Advanced_Multi_Block;
+
+/**
+ * Enqueues class.
+ */
+class Enqueues extends Plugin_Module {
+	/**
+	 * Path resolver for build directory.
+	 *
+	 * @var Plugin_Paths
+	 */
+	private Plugin_Paths $build_dir;
+
+	/**
+	 * Setup the class.
+	 *
+	 * @param string $build_path Absolute path to the build directory for all assets.
+	 */
+	public function __construct( string $build_path ) {
+		$this->build_dir = new Plugin_Paths( $build_path );
+	}
+
+	/**
+	 * Initialize the module.
+	 */
+	public function init() {
+		add_action( 'enqueue_block_editor_assets', array( $this, 'enqueue_block_assets' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_frontend_assets' ) );
+	}
+
+	/**
+	 * Enqueues the block assets for the editor.
+	 */
+	public function enqueue_block_assets() {
+		$asset_meta = $this->build_dir->get_asset_meta( 'editor-script.js' );
```

Comment:
The `Plugin_Paths` now has a helper for resolving the `*.asset.php` meta file and returning the data, if the file is present. Removes a lot of repeated boilerplate.